src/protocol.ts

```diff
@@ -72,6 +73,7 @@ export async function protocolRead(stream: Readable, { debug } = { debug: false
         console.log('protocolRead:', `read message ${message}`)
       }
 
+      removeListeners()
```

Comment:
What do you think of putting this at the top of the method, so that if anything in the body ever becomes async, there is no chance that `closeHandler` or `endHandler` will resolve before this remove is called?

Also, line 42 already checks that stream is readable by the time `readableHandler` is attached, I don't think the listener is actually needed - can it be removed and just `read` called? I think the same applies for `closeHandler` and `endHandler` by extension, based on how the protocol works I would expect that the stream is either readable with a message or already closed when `protocolRead` is called?